### PR TITLE
Adds a new martial arts style- Drunk Brawling DNM

### PR DESCRIPTION
**Pull request by optimumtact**
Partial port of https://github.com/ParadiseSS13/Paradise/pull/823, waiting on Phils chem refactor to go through before adding this.

This adds a new martial arts datum, labeled as Drunken Brawling. This
style is automatically added and removed to people via ethanol,
specifically, 10 life ticks after they start slurring their words.

The style is mostly debuffs, as there is a flat 50% chance to miss any
harm-intent hit, a 70% chance to fail to grab, and disarm is unaffected.

The actual 'features' of this style can be summarized as such:
- Custom attack verbs
  - "jab","uppercut","overhand punch",
    "drunken right hook","drunken left hook"
  - Uppercut has a 5% chance to do 14 damage
- Limit's combat abilities
  - 70% chance to fail to passively grab
  - 50% chance to miss any harm intent
  - Disarm intent is untouched.
- Added by ethanol when someone has had it into their system
  for a while
  - Removed as soon as the ethanol becomes lower than 3 units

Conflicts:
    code/modules/reagents/Chemistry-Reagents.dm

### CODE REVIEW

**Comment by optimumtact** [comment]
WHy is our interaction handling so shit? Why don't all mobs have a species datum that handles their special natures? Why don't all mobs have a martialarts datum to handle snowflake attackign types?

Why god, Why

**Comment by optimumtact** [comment]
Why can't martial arts affect weapon wielding

why do I ask myself these things?

**Comment by optimumtact** [comment]
I missed a bracket

**Comment by optimumtact** [comment]
BYLAT LIMB ORGAN PATH

File: code/datums/martial.dm [hunk 1]
```diff
@@ -134,6 +134,72 @@
...
+
+	D.apply_damage(damage, BRUTE, null, armor_block)
+	D.apply_effect(damage, STAMINA, armor_block)
+	if(D.getStaminaLoss() > 50)
+		var/knockout_prob = D.getStaminaLoss() + rand(-15,15)
+		if((D.stat != DEAD) && prob(knockout_prob))
+			D.visible_message("<span class='danger'>[A] has knocked [D] out with a [atk_verb]!</span>", \
+								"<span class='userdanger'>[A] has knocked [D] out with a [atk_verb]!</span>")
+			D.apply_effect(10,WEAKEN,armor_block)
+			D.SetSleeping(5)
```
Comment (Fox-McCloud) [inline code/datums/martial.dm]: Paralyse, not sleep...don't think a haymaker would make someone fall asleep =p

**Comment by PKPenguin321** [comment]
Would it be possible to have it so you're less likely to get weakened by a punch with this, or does this just handle outgoing damage (rather than incoming)?

**Comment by Niknakflak** [comment]
How is a drunk person even pulling off jabs and uppercuts?  In reality, it woukd probably be all wildly swinging arms/punches and stuff.

**Comment by WJohn** [comment]
In my mind drunk brawling should make you harder to hit, not flailing and incapable of doing anything even more so than now. This seems like a straight debuff.

**Comment by Nienhaus** [comment]
https://www.youtube.com/watch?v=q5BiAiuQm4U

**Comment by Boggart** [comment]
:+1: https://www.youtube.com/watch?v=74OBuMA2qEk

File: code/datums/martial.dm (hunk 1 above)
Comment (Anonus) [inline code/datums/martial.dm]: Paralyse and sleep are very nearly the same thing so it probably doesn't matter.

**Comment by optimumtact** [comment]
I don't have much time to work on this at the moment, branch will be left up if someone wants to finish it off

todo:
hook into alcohol life handle
handle what will happen when a drunken martial artist equips boxing gloves or belt (i.e which martial art has precedence, what martial art do you restore too, should you evaluate every martial art they could potentially equip?)

**Pull request closed without merging**